pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -972,6 +972,28 @@ const (
 func (r *ReconcileClusterDeployment) resolveInstallerImage(cd *hivev1.ClusterDeployment, releaseImage string, cdLog log.FieldLogger) (*reconcile.Result, error) {
 	areImagesResolved := cd.Status.InstallerImage != nil && cd.Status.CLIImage != nil
 
+	// In minimal install mode, don't even run the image resolver pod -- just trust the InstallerImageOverride
+	if minimal, err := strconv.ParseBool(cd.Annotations[constants.MinimalInstallModeAnnotation]); err == nil && minimal {
+		if !areImagesResolved {
+			// "resolve" the images via installerImageOverride
+			installerImage := cd.Spec.Provisioning.InstallerImageOverride
+			if installerImage == "" {
+				return &reconcile.Result{}, r.updateCondition(
+					cd, hivev1.InstallImagesNotResolvedCondition, corev1.ConditionTrue,
+					"MinimalInstallModeWithoutImageOverride",
+					"InstallerImageOverride is required when using minimal-install-mode annotation", cdLog)
+			}
+			cd.Status.InstallerImage = &installerImage
+			cd.Status.CLIImage = pointer.String("<NONE>")
+			// We could count on InstallImagesNotResolvedCondition being True and let updateCondition sweep up this
+			// Status().Update(), but just in case...
+			if err := r.statusUpdate(cd, cdLog); err != nil {
+				return nil, err
+			}
```

Comment:
I don't think you need to call a status update here at all, since while exiting the function you're passing the cd object to set the condition, and that method always calls a status update.

Reply:
Yeah, I flip flopped a bit on that. The condition setter only updates if something changes in the condition. Which it *should* any time we’re in this code path. But I didn’t want to rely on it.

Reply:
Ah I missed that detail. Cool.

Reply:
So to dig in a bit...

Since `r.updateCondition()` internally decides whether to `Status().Update()` based (only!) on whether the condition changed, I couldn't use the usual pattern of tracking a `changed` var or three unless I wanted to change the call spec (and all 17 existing refs) or write some other func (which would be substantially identical except for that).

So my next thought was to pre-update the condition in such a way that `r.updateCondition()` would be guaranteed to change it -- i.e. force the status, reason, or message to something bogus. Obviously this is hacky, but I had to actually start writing it to realize *how* hacky :disappointed:.

However, I dislike the noted failure mode (if the first update succeeds but the second fails, we'll never hit this code path again, and the status condition will stay wrong forever) enough that I'm reconsidering the horrible hack.

...unless you have other ideas?

Reply:
These options are very hacky and at that point, we are making it confusing. `updateCondition` was written because we update so many cd conditions within the controller that it wraps the logic of setting the condition and calling the update to make it simpler for the caller Reconcile to safely exit. Since this is a special case, I recommend you directly call the controllerutils setter and explicitly update here as you exit.

Reply:
Yeah, that's fair. Will fix. Thanks!

Reply:
Done.